lib/services/account-api-service.js

```diff
@@ -0,0 +1,193 @@
+                acl.addRoleParents('Administrator', ['Read', 'Write']),
+                acl.addRoleParents('ReadOnly', ['Read']),
+
+                // Redfish 1.0 fixed privileges
+                acl.addRoleParents('Administrator', ['Login','ConfigureManager','ConfigureUsers']),
+                acl.addRoleParents('Administrator', ['ConfigureComponents','ConfigureSelf']),
+                acl.addRoleParents('Operator', ['Login','ConfigureComponents','ConfigureSelf']),
+                acl.addRoleParents('ReadOnly', ['Login', 'ConfigureSelf'])
+            ]);
+        });
+    };
+
+    AccountApiService.prototype.aclMethod = function(name) {
+        var args = Array.prototype.slice.call(arguments, 1);
+        return this.acl[name].apply(this.acl, args);
+    };
+
+    AccountApiService.prototype.listUsers = function() {
+        var self = this;
+        return Promise.map(waterline.localusers.find({}), function(user) {
+            return Promise.props({
+                id: user.id,
+                username: user.username,
+                role: user.role,
+                privileges: self.aclMethod('_rolesParents', [user.role])
+            });
+        });
+    };
+
+    AccountApiService.prototype.getUserByName = function(username) {
+        var self = this;
+        return Promise.try(function() {
+            assert.string(username);
+            return waterline.localusers.findOne({username: username});
+        })
+        .then(function(user) {
+            if(!user) {
+                throw new Errors.NotFoundError();
+            }
+            return Promise.props({
+                id: user.id,
+                username: user.username,
+                role: user.role,
+                privileges: self.aclMethod('_rolesParents', [user.role])
+            });
+        });
+    };
+
+    AccountApiService.prototype.createUser = function(user) {
+        var self = this;
+        return Promise.try(function() {
+            assert.string(user.username);
+            return waterline.localusers.findOne({username: user.username});
+        })
+        .then(function(entry) {
+            if(entry) {
+                // TODO: Replace with HttpError after errors are refactored
+                var err = new Errors.BaseError('the user already exists');
+                err.status = 409;
+                throw err;
+            }
+            return waterline.localusers.create({
+                username: user.username,
+                password: user.password,
+                role: user.role
+            });
+        })
+        .then(function(entry) {
+            return self.aclMethod('addUserRoles', entry.username, entry.role);
+        })
+        .then(function() {
+            return self.getUserByName(user.username);
+        });
+    };
+
+    AccountApiService.prototype.modifyUserByName = function(name, info) {
+        var self = this;
+        return Promise.try(function() {
```

Comment:
I see some duplicated code for these APIs, following code are shared by most APIs, I suggest wrap the code to a function:
```
        return Promise.try(function() {
            assert.string(username);
            return waterline.localusers.findOne({username: username});
        })
        .then(function(user) {
            if(!user) {
                throw new Errors.NotFoundError();
            }
```